File: frontend/lib/models/lead.dart
import 'package:frontend/models/User.dart';
import 'package:frontend/models/campaign.dart';
import 'package:frontend/models/meeting.dart';

class Lead {
  int? id;
  String? leadName;
  String? leadAddress;
  String? leadCity;
  String? leadState;
  String? leadZip;
  String? leadCountry;
  String? leadPhone;
  String? leadEmail;
  String? leadFax;
  String? leadWebsite;
  String? leadNotes;
  DateTime? created;
  DateTime? updated;
  String? createdBy;
  User? createdByUser;
  List<Campaign>? campaigns;
  List<Meeting>? meetings;

  Lead({
    this.id,
    this.leadName,
    this.leadAddress,
    this.leadCity,
    this.leadState,
    this.leadZip,
    this.leadCountry,
    this.leadPhone,
    this.leadEmail,
    this.leadFax,
    this.leadWebsite,
    this.leadNotes,
    this.created,
    this.updated,
    this.createdBy,
    this.createdByUser,
    this.campaigns,
    this.meetings,
  });

  factory Lead.fromJson(Map<String, dynamic> json) {
    return Lead(
      id: json['id'],
      leadName: json['leadName'],
      leadAddress: json['leadAddress'],
      leadCity: json['leadCity'],
      leadState: json['leadState'],
      leadZip: json['leadZip'],
      leadCountry: json['leadCountry'],
      leadPhone: json['leadPhone'],
      leadEmail: json['leadEmail'],
      leadFax: json['leadFax'],
      leadWebsite: json['leadWebsite'],
      leadNotes: json['leadNotes'],
      created: DateTime.parse(json['created']),
      updated: DateTime.parse(json['updated']),
      createdBy: json['createdBy'],
      createdByUser: json['createdByUser'] != null
          ? User.fromJson(json['createdByUser'])
          : null,
      campaigns: (json['campaigns'] as List?)
          ?.map((item) => Campaign.fromJson(item))
          .toList(),
      meetings: (json['meetings'] as List?)
          ?.map((item) => Meeting.fromJson(item))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'leadName': leadName,
      'leadAddress': leadAddress,
      'leadCity': leadCity,
      'leadState': leadState,
      'leadZip': leadZip,
      'leadCountry': leadCountry,
      'leadPhone': leadPhone,
      'leadEmail': leadEmail,
      'leadFax': leadFax,
      'leadWebsite': leadWebsite,
      'leadNotes': leadNotes,
      'created': created?.toIso8601String(),
      'updated': updated?.toIso8601String(),
      'createdBy': createdBy,
      'createdByUser': createdByUser?.toJson(),
      'campaigns': campaigns?.map((e) => e.toJson()).toList(),
      'meetings': meetings?.map((e) => e.toJson()).toList(),
    };
  }
}

File: backend/backend/Areas/Main/Models/Lead.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Areas.Communication.Models;
using backend.Areas.Identity.Models;

namespace backend.Areas.Main.Models;

public class Lead
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    public string LeadName { get; set; }
    public string? LeadAddress { get; set; }
    public string? LeadCity { get; set; }
    public string? LeadState { get; set; }
    public string? LeadZip { get; set; }
    public string? LeadCountry { get; set; }
    public string LeadPhone { get; set; }
    public string? LeadEmail { get; set; }
    public string? LeadFax { get; set; }
    public string? LeadWebsite { get; set; }
    public DateTime Created { get; set; }
    public DateTime? Updated { get; set; }
    
    public string? CreatedBy { get; set; }
    [ForeignKey("CreatedBy")] 
    public User? CreatedByUser { get; set; }
    public virtual IEnumerable<LeadNotes>? LeadNotes{ get; set; }
    public virtual IEnumerable<Campaign>? Campaigns { get; set; }
    public virtual IEnumerable<Meeting>? Meetings { get; set; }
    public virtual IEnumerable<LeadTask>? LeadTasks { get; set; }
}